lib/vnet/diag/routeconflict_test.go

```diff
@@ -0,0 +1,317 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package diag
+
+import (
+	"context"
+	"net"
+	"net/netip"
+	"testing"
+
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+)
+
+const vnetIface = "utun4"
+const quuxIface = "utun5"
+const vnetIfaceIndex = 1
+const quuxIfaceIndex = 2
```

Comment:
`quux` is used instead of `other` or `their` or whatever so that the variables line up with `vnetIface` and `vnetIfaceIndex` when we define multiple routes in a row.